base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync.rs

```diff
@@ -130,6 +130,12 @@ impl HorizonStateSync {
                     randomx_vm_cnt,
                     randomx_vm_flags,
                 });
+                if let Err(e) = shared.db.reset_blockchain_databases().await {
```

Comment:
I think we'll need to determine this from after header sync - so once we know the size of the reorg from the headers. We'll know if it's beyond the pruning horizon and can reset the db.